llap-server/src/test/org/apache/hadoop/hive/llap/daemon/impl/TestLlapLoadGeneratorService.java

```diff
@@ -40,7 +44,32 @@ public void testLoadGenerator() throws InterruptedException, UnknownHostExceptio
 
     service.init(conf);
     service.start();
-    Thread.sleep(10000);
+    Thread.sleep(5000);
+    assertEquals("The number of threads is not correct",
+        Runtime.getRuntime().availableProcessors(), service.threads.length);
+    for(int i = 0; i < service.threads.length; i++) {
+      assertTrue("The thread [" + i + "] should be alive", service.threads[i].isAlive());
+    }
+    service.stop();
+    Thread.sleep(5000);
+    for(int i = 0; i < service.threads.length; i++) {
+      Thread.State state = service.threads[i].getState();
```

Comment:
This seems to be unused.

Reply:
I think this is used. The service is stopped and it interrupts the load generating threads.
After that we check if the threads are really stopped or not